### ISSUE TITLE
codegen on pull: `amplify codegen models` does not work with multiple frontends

### ISSUE THREAD

**Issue by ataibarkai** [aws-amplify/amplify-cli#5754]
>**Note**: If your issue/bug is regarding the AWS Amplify Console service, please log it in the 
[Amplify Console GitHub Issue Tracker](https://github.com/aws-amplify/amplify-console/issues)

**Describe the bug**
`amplify codegen models` does not work in secondary frontend project setup according to the  [amplify docs](https://docs.amplify.aws/cli/teams/multi-frontend) -- i.e. with specifying `Do you plan on modifying this backend? No`

**Amplify CLI Version**
You can use `amplify -v` to check the amplify cli version on your system
4.31.1

**To Reproduce**
- Setup an amplify project with a GraphQL API.

- Create a secondary iOS frontend using the workflow described in the [amplify docs](https://docs.amplify.aws/cli/teams/multi-frontend). (In particular,  Respond with `NO` to the prompt `Do you plan on modifying this backend?`)

- Run `amplify codegen models` in its root folder

- Observe failure:
```
You are not working inside a valid Amplify project.
Use 'amplify init' in the root of your app directory to initialize your project, or 'amplify pull' to pull down an existing project.
```

- You can run `amplify pull` and go through the flow again, there will be no change.

- If instead I pull the amplify project and specify 
`? Do you plan on modifying this backend? YES`
instead of the `NO` recommended in the multi-frontend docs, `amplify codegen models` **does** work.

**Expected behavior**
`amplify codegen models` should run successfully when specifying  `Do you plan on modifying this backend? NO`

**Screenshots**

**Desktop (please complete the following information):**
 - OS: [e.g. Mac/Windows/Ubuntu] macOS 10.15.7
 - Node Version. node v12.18.3

**Additional context**

**Comment by akshbhu** [comment]
Hi @ataibarkai 

Currently to generate models , you have select the option'Y' when ask about modifying backend. 
I am marking this as enhancement. May I know one thing, Is there a specific reason for choosing  'N' when modifying backend in your case?

**Comment by ataibarkai** [comment]
Thanks for the information @akshbhu.

>  Is there a specific reason for choosing 'N' when modifying backend in your case?

I was simply following [Amplify's multi-frontend docs](https://docs.amplify.aws/cli/teams/multi-frontend) (which suggest choosing 'No' for that prompt), but I also agree with this suggestion: I have multiple frontends but I only want to define the backend in one place, to avoid the potential for conflict and synchronization issues.

**Comment by amuresia** [comment]
+1

**Comment by aldegoeij** [comment]
+1 

multi-frontend basically does not work as marking N for change backend does not copy any config to the project, so none of the amplify commands work. 

The pull does retrieve the models though, but only once. Also seems like `aws-exports.js` does not get updated after initial pull.

Went the manual route for the frontend, keeping models with backend.

**Comment by kaustavghosh06** [comment]
Closing as duplicate of #5251
